README.md

```diff
@@ -1,31 +1,120 @@
-# Azure Spring Apps Sample - Simple Todo App
+---
+page_type: sample
+languages:
+- azdeveloper
+- java
+- bicep
+  products:
+- azure
+- azure-spring-apps
+- azure-postgresql
+- azure-key-vault
+- azure-pipelines
+- ms-build-openjdk
+  urlFragment: todo-java-postgresql-asa
+  name: React Web App with Java API and PostgreSQL - Flexible Server on Azure App Service
+  description: A complete ToDo app on Azure Spring Apps with Java API and Azure Database for PostgreSQL flexible server for storage. Uses Azure Developer CLI (azd) to build, deploy, and run
+---
+<!-- YAML front-matter schema: https://review.learn.microsoft.com/en-us/help/contribute/samples/process/onboarding?branch=main#supported-metadata-fields-for-readmemd -->
+# React Web App with Java API and PostgreSQL - Flexible Server on Azure App Service
 
-There are 2 branches about this repository:
-1. [quickstart](https://github.com/Azure-Samples/ASA-Samples-Web-Application/blob/quickstart/README.md). This branch is used to teach you to run [spring web application](https://spring.io/web-applications) in [Azure Spring Apps](https://learn.microsoft.com/en-us/azure/spring-apps/overview) (ASA) in a short time.
-2. [reference-architecture](https://github.com/Azure-Samples/ASA-Samples-Web-Application/blob/reference-architecture/README.md). This branch give the reference architecture (best practice) of running spring web application in Azure Spring Apps.
+A blueprint for getting a React web app with a Java API and a PostgreSQL - Flexible Server on Azure. The blueprint includes sample application code (a ToDo web app) which can be removed and replaced with your own application code. Add your own source code and leverage the Infrastructure as Code assets (written in Bicep) to get up and running quickly. This architecture is for running containerized apps or microservices on a serverless platform.
+
+Let's jump in and get this up and running in Azure. When you are finished, you will have a fully functional web app deployed to the cloud. In later steps, you'll see how to setup a pipeline and run the application.
+
+!["Screenshot of deployed ToDo app"](assets/web.png)
+
+<sup>Screenshot of the deployed ToDo app</sup>
 
 ## Prerequisites
 
-- Java 17 or later
+The following prerequisites are required to use this application. Please ensure that you have them all installed locally.
+
+- [Azure Developer CLI](https://aka.ms/azd-install)
+- [Java 17 or later](https://learn.microsoft.com/en-us/java/openjdk/install) - for API backend
+- [Node.js with npm (16.13.1+)](https://nodejs.org/) - for the Web frontend
+- [Docker](https://docs.docker.com/get-docker/)
```

Comment:
I guess we should add PostgreSQL Command line tools?

Reply:
Since we switched to use deployment script to create the database user, thus, we can omit this requirement?

Reply:
> Since we switched to use deployment script to create the database user, thus, we can omit this requirement?

You are correct, we don't need it.